### PR TITLE
rz-test: support exclude some directories

### LINKED ISSUE #336: Allow to disable debug tests in `rz-test` command line
Currently `rz-test` either runs everything or you need to specify the particular directory/test to run.
But there is no way to run **everything** excluding a particular directory/category.

Biggest consumer case of this feature would be running all tests except debug ones.

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

closes https://github.com/rizinorg/rizin/issues/336
...

### CODE REVIEW

**Comment by XVilka** [comment]
Please also add ignoring debug tests here: https://github.com/rizinorg/rizin/blob/dev/.github/workflows/ci.yml#L394

**Comment by XVilka** [comment]
Please also resend this from the branch name starting with `dist-`

File: binrz/rz-test/rz-test.c [hunk 1]
```diff
@@ -409,6 +420,23 @@ int rz_test_main(int argc, const char **argv) {
 		}
 	}
 
+	// filter out except_dir
+	if (rz_pvector_len(except_dir)) {
```
Comment (ret2libc) [inline binrz/rz-test/rz-test.c:424]:
```suggestion
	if (!rz_pvector_empty(except_dir)) {
```

File: binrz/rz-test/rz-test.c [hunk 2]
```diff
@@ -409,6 +420,23 @@ int rz_test_main(int argc, const char **argv) {
...
+		rz_pvector_foreach (except_dir, it) {
+			char *p = rz_file_abspath_rel(cwd, (char *)*it);
+			for (ut32 i = 0; i < rz_pvector_len(&state.db->tests);) {
+				RzTest *test = rz_pvector_at(&state.db->tests, i);
+				if (rz_str_startswith(test->path, p)) {
+					rz_test_test_free(test);
+					rz_pvector_remove_at(&state.db->tests, i);
+					continue;
+				}
+				i++;
```
Comment (ret2libc) [inline binrz/rz-test/rz-test.c:435]: why is this here and not in the `for`?

**Comment by PeiweiHu** [comment]
> Please also resend this from the branch name starting with `dist-`

Sorry, I don't get it. Do you mean close this one, then apply the patch on a branch starting with 'dist-' and also pull request on that 'dist-' branch? There are lots of 'dist-' branch. Which one should I choose?

File: binrz/rz-test/rz-test.c (hunk 2 above)
Comment (PeiweiHu) [reply binrz/rz-test/rz-test.c:435]: Because no need to increment `i` if remove happens. It also works if put this in for, then performs `i--` when remove happens.

File: binrz/rz-test/rz-test.c (hunk 1 above)
Comment (PeiweiHu) [reply binrz/rz-test/rz-test.c:424]: Thanks for your review. I will change later.

**Comment by ret2libc** [comment]
> > Please also resend this from the branch name starting with `dist-`
> 
> Sorry, I don't get it. Do you mean close this one, then apply the patch on a branch starting with 'dist-' and also pull request on that 'dist-' branch? There are lots of 'dist-' branch. Which one should I choose?

Close this PR. Create a new branch with the name starting with `dist-`. For example `git checkout -b dist-patch-exclude-dirs-rztest` then `git push` then create the PR against rizinorg:dev but from the dist-patch-exclude-dirs-rztest branch. The name of the branch is used to trigger additional CI jobs.

**Comment by PeiweiHu** [comment]
> > > Please also resend this from the branch name starting with `dist-`
> > 
> > 
> > Sorry, I don't get it. Do you mean close this one, then apply the patch on a branch starting with 'dist-' and also pull request on that 'dist-' branch? There are lots of 'dist-' branch. Which one should I choose?
> 
> Close this PR. Create a new branch with the name starting with `dist-`. For example `git checkout -b dist-patch-exclude-dirs-rztest` then `git push` then create the PR against rizinorg:dev but from the dist-patch-exclude-dirs-rztest branch. The name of the branch is used to trigger additional CI jobs.

Thanks. I got it.
